### PR TITLE
Install torchrec/fbgemm from source in CI

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106808

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

**Comment by ezyang** [comment]
@davidberard98 @xuzhao9 I cc'ed you because y'all were involved in https://github.com/pytorch/benchmark/pull/1427

File: .ci/pytorch/common_utils.sh [hunk 1]
```diff
@@ -158,6 +158,27 @@ function install_torchvision() {
...
+  pip_uninstall torchrec-nightly
+  pip_uninstall fbgemm-gpu-nightly
+  function doit() {
+    pip_install --no-use-pep517 --user "git+https://github.com/pytorch/FBGEMM.git@${commit}#egg=fbgemm-gpu&subdirectory=fbgemm_gpu"
+    pip_install --no-use-pep517 --user "git+https://github.com/pytorch/torchrec.git@${commit}"
+  }
+  if [[ "$1" == "cuda" ]]; then
+    # TODO: This is better to be passed as a parameter from _linux-test workflow
+    # so that it can be consistent with what is set in build
+    TORCH_CUDA_ARCH_LIST="8.0;8.6" doit
```
Comment (malfet) [inline .ci/pytorch/common_utils.sh:175]: Hmm, there is already `torch.cuda.get_gencode_flags()`, but worst case you can do something like
```suggestion
    TORCH_CUDA_ARCH_LIST=$(python3 -c "import torch;props=torch.cuda.get_device_properties(0);print(f'{props.major}.{props.minor}')") doit
```

**Review by malfet — APPROVED** [review]

**Comment by xuzhao9** [comment]
Are any of the model in the CI currently using fbgemm so far? In TB we used to have torchrec_dlrm which uses fbgemm, but it was moved to canary_models later due to flakiness in the fbgemm_gpu_nightly package. But this PR seems to be always compiling fbgemm so I guess this is fine.

**Comment by ezyang** [comment]
> Are any of the model in the CI currently using fbgemm so far? In TB we used to have torchrec_dlrm which uses fbgemm, but it was moved to canary_models later due to flakiness in the fbgemm_gpu_nightly package. But this PR seems to be always compiling fbgemm so I guess this is fine.

dlrm and torchrec_dlrm. And yes, the point of this change is to get rid of the instability from using the nightly packages!

File: .ci/pytorch/common_utils.sh (hunk 1 above)
Comment (ezyang) [reply .ci/pytorch/common_utils.sh:175]: I cargo culted this from the torchaudio instructions. Let me try your suggestion.

Comment (ezyang) [reply .ci/pytorch/common_utils.sh:175]: Hmm, I don't think your suggestion is right. What if we're doing a build from a CPU runner? You won't specify arch list in that case.

**Comment by xuzhao9** [comment]
> @xuzhao9 🤔 does dlrm even use torchrec

No, I don't think the "dlrm" model in torchbench (https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/dlrm/__init__.py) uses torchrec, because it is dlrm v1 (the oldest version of dlrm).

The `torchrec_dlrm` model in torchbench (https://github.com/pytorch/benchmark/tree/main/torchbenchmark/canary_models/torchrec_dlrm) uses torchrec and fbgemm, but we moved it to canary_models because of flakiness of torchrec_nightly and fbgemm_gpu_nightly.

**Review by xuzhao9 — APPROVED** [review]
LGTM

File: .ci/pytorch/common_utils.sh (hunk 1 above)
Comment (ezyang) [reply .ci/pytorch/common_utils.sh]: I just deleted this entirely and pray

**Pull request closed without merging**